File: src/main/java/com/impact/mods/gregtech/items/tools/behaviour/Behaviour_OreProbe.java
package com.impact.mods.gregtech.items.tools.behaviour;

import com.impact.common.oregeneration.OreGenerator;
import com.impact.common.oregeneration.OreVein;
import com.impact.util.Utilits;
import gregtech.api.enums.Materials;
import gregtech.api.enums.OrePrefixes;
import gregtech.api.interfaces.IItemBehaviour;
import gregtech.api.items.GT_MetaBase_Item;
import gregtech.api.util.GT_LanguageManager;
import gregtech.api.util.GT_Utility;
import gregtech.common.items.behaviors.Behaviour_None;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.item.ItemStack;
import net.minecraft.util.EnumChatFormatting;
import net.minecraft.util.StatCollector;
import net.minecraft.world.World;
import net.minecraft.world.chunk.Chunk;
import net.minecraftforge.oredict.OreDictionary;

public class Behaviour_OreProbe extends Behaviour_None {
	
	public static final IItemBehaviour<GT_MetaBase_Item> INSTANCE = new Behaviour_OreProbe();
	
	public boolean onItemUseFirst(GT_MetaBase_Item aItem, ItemStack aStack, EntityPlayer aPlayer, World aWorld, int aX, int aY, int aZ, int aSide, float hitX, float hitY, float hitZ) {
		if (aPlayer instanceof EntityPlayerMP) {
			Chunk ch = aWorld.getChunkFromBlockCoords(aX, aZ);
			OreVein oreVein = OreGenerator.getOreVein(ch, 0);
			if (oreVein != null) {
				int countOres = oreVein.ores.size();
				ItemStack is = oreVein.ores.get(Utilits.getRandom(0, countOres - 1));
				
				GT_Utility.sendChatToPlayer(aPlayer, "Presumably there is Ore here: " + EnumChatFormatting.YELLOW + Utilits.translateGTItemStack(is));
			} else {
				GT_Utility.sendChatToPlayer(aPlayer, "Not found Ores");
			}
			
			return true;
		}
		return false;
	}
}

File: src/main/java/com/impact/mods/gregtech/items/tools/behaviour/BehaviourOreProbe.kt
package com.impact.mods.gregtech.items.tools.behaviour

import com.impact.util.Utilits
import gregtech.api.interfaces.IItemBehaviour
import gregtech.api.items.GT_MetaBase_Item
import gregtech.api.util.GT_Utility
import gregtech.common.items.behaviors.Behaviour_None
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.item.ItemStack
import net.minecraft.util.EnumChatFormatting
import net.minecraft.world.World
import space.gtimpact.virtual_world.api.VirtualAPI
import space.gtimpact.virtual_world.api.core.WorldPos
import space.gtimpact.virtual_world.api.services.scanning.ScanMode

class BehaviourOreProbe : Behaviour_None() {

    override fun onItemUseFirst(
        aItem: GT_MetaBase_Item, aStack: ItemStack, aPlayer: EntityPlayer,
        aWorld: World, aX: Int, aY: Int, aZ: Int, aSide: Int, hitX: Float,
        hitY: Float, hitZ: Float
    ): Boolean {
        if (aPlayer is EntityPlayerMP) {
            val ch = aWorld.getChunkFromBlockCoords(aX, aZ)

            val result = VirtualAPI.scanning.scanOreAroundBlock(
                mode = ScanMode.WITH_AMOUNT,
                layerIndex = 0,
                radiusVeins = 0,
                dimensionId = ch.worldObj.provider.dimensionId,
                centerBlockPos = WorldPos(x = aX, z = aZ),
            ).results.firstOrNull()

            if (result != null) {
                VirtualAPI.scannerManager.scanOres(
                    player = aPlayer,
                    mode = ScanMode.BOUNDARY_ONLY,
                    layer = 0,
                    dimensionId = ch.worldObj.provider.dimensionId,
                    radiusVeins = 0,
                )
                val labelVein = Utilits.translateGTItemStack(result.ore.ores.first().stack)
                GT_Utility.sendChatToPlayer(aPlayer, "Presumably there is Ore here: " + EnumChatFormatting.YELLOW + labelVein)
            } else {
                GT_Utility.sendChatToPlayer(aPlayer, "Not found Ores")
            }

            return true
        }
        return false
    }

    companion object {
        @JvmField
        val INSTANCE: IItemBehaviour<GT_MetaBase_Item> = BehaviourOreProbe()
    }
}